zgrab2/processing.go

```diff
@@ -0,0 +1,126 @@
+package zgrab2
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"net"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Grab struct {
+	IP     *string                   `json:"ip,omitempty"`
```

Comment:
omitempty for json marshaling only works if the type is set to nil. A string can't be set to nil and the empty string isn't empty according to omitempty. 
It does get sent through a channel